dtfs-central-api/api-tests/v1/gef-deals/add-min-activities.api-test.js

```diff
@@ -0,0 +1,413 @@
+      expect(receivedDate).toEqual(expectedDate);
+    });
+
+    it('should correctly return author', () => {
+      const expectedUser = {
+        firstName: 'tester',
+        lastName: 'testing',
+        _id: 12345,
+      };
+      expect(result.author).toEqual(expectedUser);
+    });
+
+    it('should correctly return text', () => {
+      expect(result.text).toEqual('test123');
+    });
+
+    it('should correctly return label', () => {
+      expect(result.label).toEqual('Manual Inclusion Application submitted to UKEF');
+    });
+
+    it('should correctly return html', () => {
+      expect(result.html).toEqual('facility');
+    });
+
+    it('should correctly return maker', () => {
+      expect(result.maker).toEqual(maker);
+    });
+
+    it('should correctly return checker', () => {
+      expect(result.checker).toEqual(checker);
+    });
+
+    it('should correctly return type', () => {
+      expect(result.facilityType).toEqual('Cash facility');
+    });
+
+    it('should correctly return facilityID', () => {
+      expect(result.facilityID).toEqual(123456);
+    });
+  });
+
+  describe('should correctly return populated object where facility undefined', () => {
+    const generatorObj = {
+      type: applicationType,
+      user,
+      activityType,
+      activityText,
+      activityHTML,
+      facility: null,
+      checker,
+      maker,
+    };
+    // ensures the returned object is properly generated with required fields
+    const result = portalActivityGenerator(generatorObj);
+    it('should correctly return type', () => {
+      expect(result.type).toEqual('NOTICE');
+    });
+
+    it('should correctly return timestamp', () => {
+    // matches date as timestamps may be seconds off
+      const receivedDate = format(fromUnixTime(result.timestamp), 'dd-MMMM-yyyy');
+      const expectedDate = format(new Date(), 'dd-MMMM-yyyy');
+      expect(receivedDate).toEqual(expectedDate);
+    });
+
+    it('should correctly return author', () => {
+      const expectedUser = {
+        firstName: 'tester',
+        lastName: 'testing',
+        _id: 12345,
+      };
+      expect(result.author).toEqual(expectedUser);
+    });
+
+    it('should correctly return text', () => {
+      expect(result.text).toEqual('test123');
+    });
+
+    it('should correctly return label', () => {
+      expect(result.label).toEqual('Manual Inclusion Application submitted to UKEF');
+    });
+
+    it('should correctly return html', () => {
+      expect(result.html).toEqual('facility');
+    });
+
+    it('should correctly return maker', () => {
+      expect(result.maker).toEqual(maker);
+    });
+
+    it('should correctly return checker', () => {
+      expect(result.checker).toEqual(checker);
+    });
+
+    it('should correctly return facilityType', () => {
+      expect(result.facilityType).toEqual('');
+    });
+
+    it('should correctly return facilityID', () => {
+      expect(result.facilityID).toEqual('');
```

Comment:
can we combine all these `expect` into one `it` statement? there are quite a few of them that do expect 1 thing

Reply:
I had done this before after speaking to Abhi on a previous ticket so its easier to see exactly where something is going wrong, I can change it back if you prefer